tempesta_fw/http.c

```diff
@@ -748,9 +748,18 @@ tfw_http_adjust_resp(TfwHttpResp *resp, TfwHttpReq *req)
 		return r;
 
 	r = tfw_http_add_hdr_via(hm);
-	if (r)
-		return r;
-
+	if (resp->flags & TFW_HTTP_RESP_STALE) {
+#define S_WARN_110_N "Warning"
+#define S_WARN_110_V "110 - Response is stale"
+		r = tfw_http_msg_hdr_xfrm(hm, S_WARN_110_N,
+ 					  SLEN(S_WARN_110_N), S_WARN_110_V,
+					  SLEN(S_WARN_110_V),
+					  TFW_HTTP_HDR_RAW, 0);
```

Comment:
Why `tfw_http_msg_hdr_xfrm()` instead of `tfw_http_msg_hdr_add()`? It seems there is no problem if upstream server returns response with the warning and we retrieve it from the cache. Meantime we don't store the adjusted response in the cache and we can ovoid the header lookup overhead if we use `tfw_http_msg_hdr_add()`.

By the way, it'd be better to place `SLEN(S_WARN_110_N)` on the same line with `S_WARN_110_N`.